src/data/mod.rs

```diff
@@ -33,35 +33,77 @@ impl Context {
+            // Make note of whether we've found prereleases or stable releases yet
+            if is_prerelease {
+                if !has_prereleases {
+                    let msg = "Detected pre-releases...";
+                    Message::new(MessageType::Info, msg).print();
+                    has_prereleases = true;
+                }
+            } else if latest_stable_release.is_none() {
+                latest_stable_release = Some(idx);
             }
-            if !found_latest_dist_release && gh_release.has_dist_manifest() {
+
+            // Special handling of dist-manifest.json
+            if release.manifest.is_some() {
                 if cargo_dist {
-                    let release = Release::new(gh_release.clone(), cargo_dist)?;
-                    if let Some(manifest) = release.manifest {
-                        latest_dist_release = Some(DistRelease {
-                            manifest,
-                            source: gh_release.clone(),
-                        });
+                    // cargo-dist is enabled, so we want to find the latest stable release
+                    // or, failing that, the latest prerelease.
+                    if is_prerelease {
+                        if latest_dist_prerelease.is_none() && latest_dist_release.is_none() {
+                            latest_dist_prerelease = Some(idx);
+                        }
+                    } else if latest_dist_release.is_none() {
+                        latest_dist_release = Some(idx);
                     }
-                    found_latest_dist_release = latest_dist_release.is_some();
                 } else if !warned {
+                    // We found a dist-manifest but they didn't enable cargo-dist support, encourage them to do so
                     let msg = "You have not configured cargo-dist yet we detected dist-manifests in your releases. Is this intended?";
                     Message::new(MessageType::Warning, msg).print();
                     warned = true;
                 }
             }
-            all.push(Release::new(gh_release, cargo_dist)?)
         }
-        Ok((all, has_prereleases, latest_dist_release))
+
+        // If we found any kind of stable release, then throw out any prerelease fallbacks.
+        // This prevents us recommending an unstable dist-release over a stable non-dist-release.
+        if latest_stable_release.is_some() {
+            latest_dist_prerelease = None;
+        }
+
+        // If we found a stable cargo-dist release, but there's even newer stable releases
+        // that don't use cargo-dist, we're going to prefer the cargo-dist one, but we should
+        // warn the user that things are wonky
+        if latest_dist_release.is_some() && latest_dist_release != latest_stable_release {
+            let msg = "You have newer stable Github Releases than your latest cargo-dist Release. Is this intended? (We're going to prefer the cargo-dist one.)";
```

Comment:
i could imagine it'd be useful to say the release names perhaps, e.g. "name@version" - it would make it easier to know if things were correct or not

Reply:
went with tag-name